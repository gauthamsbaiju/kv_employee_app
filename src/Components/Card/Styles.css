.card-wrapper{
    background-color: white;
    height: 232px;
    width: calc(100% - 326px);
    margin-top: 26px;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
padding: 34px;
}