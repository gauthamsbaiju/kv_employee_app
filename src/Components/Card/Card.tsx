import React from 'react';
import './Styles.css';
import EmployeeDetails from '../EmployeeDetails/EmployeeDetails';
// import employees from '../../Dummy/Employees';

export type CardPropsType = {
  emp: {
    id: number;
    name: string;
    joiningDate: string;
    experience: number;
    isActive: boolean;
    role: string;
  };
};

const Card: React.FC<CardPropsType> = (props) => {
  console.log(props.emp);
  const details = Object.entries(props.emp);

  details.map((item) => console.log(item));

  return (
    <div className='card-wrapper' data-testid='card-test'>
      {/* {details.map((item, index) => (
        <EmployeeDetails lable={item[0]} value={item[1]} key={index} />
      ))} */}
      <EmployeeDetails lable='Employee Name' value={props.emp.name} type='text' />
      <EmployeeDetails lable='Joining Date' value={props.emp.joiningDate} type='text' />
      <EmployeeDetails lable='Experience' value={props.emp.experience} type='text' />
      <EmployeeDetails lable='Role' value={props.emp.role} type='text' />
      <EmployeeDetails lable='Status' value={props.emp.isActive} type='status' />
      <EmployeeDetails lable='Employee ID' value={props.emp.id} type='text' />
    </div>
  );
};

export default Card;
