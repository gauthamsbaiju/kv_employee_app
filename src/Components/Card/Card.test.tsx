import { render, screen } from '@testing-library/react';
import Card, { CardPropsType } from './Card';
import '@testing-library/jest-dom';

describe('Test for employee details card', () => {
  //   test('Test for card rendered', () => {
  //     const CardProps: CardPropsType = {
  //       emp: {
  //         id: 1,
  //         name: 'Abc',
  //         joiningDate: '12/12/12',
  //         experience: 4,
  //         isActive: true,
  //         role: 'user'
  //       }
  //     };

  //     render(<Card {...CardProps}></Card>);
  //     const element = screen.getByTestId('card-test');

  //     if (element)
  //       expect(element).toHaveAttribute('emp', {
  //         id: 1,
  //         name: 'Abc',
  //         joiningDate: '12/12/12',
  //         experience: 4,
  //         isActive: true,
  //         role: 'user'
  //       });
  //   });

  test('If snapshot rendered', () => {
    const CardProps: CardPropsType = {
      emp: {
        id: 1,
        name: 'Abc',
        joiningDate: '12/12/12',
        experience: 4,
        isActive: true,
        role: 'user'
      }
    };

    render(<Card {...CardProps} />);
    const element = screen.getByTestId('card-test');

    expect(element).toMatchSnapshot();
  });
});
