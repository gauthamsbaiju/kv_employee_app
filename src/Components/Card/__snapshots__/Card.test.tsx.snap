// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for employee details card If snapshot rendered 1`] = `
<div
  class="card-wrapper"
  data-testid="card-test"
>
  <div
    class="details-wrapper"
  >
    <div
      class="details-label"
      data-testid="employeeDetailsLabel-test"
    >
      Employee Name
    </div>
    <div
      data-testid="employeeDetailsValue-test"
    >
      Abc
    </div>
  </div>
  <div
    class="details-wrapper"
  >
    <div
      class="details-label"
      data-testid="employeeDetailsLabel-test"
    >
      Joining Date
    </div>
    <div
      data-testid="employeeDetailsValue-test"
    >
      12/12/12
    </div>
  </div>
  <div
    class="details-wrapper"
  >
    <div
      class="details-label"
      data-testid="employeeDetailsLabel-test"
    >
      Experience
    </div>
    <div
      data-testid="employeeDetailsValue-test"
    >
      4
    </div>
  </div>
  <div
    class="details-wrapper"
  >
    <div
      class="details-label"
      data-testid="employeeDetailsLabel-test"
    >
      Role
    </div>
    <div
      data-testid="employeeDetailsValue-test"
    >
      user
    </div>
  </div>
  <div
    class="details-wrapper"
  >
    <div
      class="details-label"
      data-testid="employeeDetailsLabel-test"
    >
      Status
    </div>
    <div
      class="active"
      data-testid="status-test"
    >
      active
    </div>
    <div
      data-testid="employeeDetailsValue-test"
    />
  </div>
  <div
    class="details-wrapper"
  >
    <div
      class="details-label"
      data-testid="employeeDetailsLabel-test"
    >
      Employee ID
    </div>
    <div
      data-testid="employeeDetailsValue-test"
    >
      1
    </div>
  </div>
</div>
`;
